numpy/lib/tests/test__iotools.py

```diff
@@ -133,10 +133,9 @@ def test_validate_wo_names(self):
 
 
 def _bytes_to_date(s):
-    if sys.version_info[0] >= 3:
-        return date(*time.strptime(s.decode('latin1'), "%Y-%m-%d")[:3])
-    else:
-        return date(*time.strptime(s, "%Y-%m-%d")[:3])
+    if type(s) == bytes:
+        s = s.decode("latin1")
+    return date(*time.strptime(s, "%Y-%m-%d")[:3])
```

Comment:
Shouldn't the contract be that `s` is now always `unicode`, and so the decode is unconditional?

Reply:
`_bytes_to_date` is a helper function called by several other tests, so it probably needs to do the check. Mind that we are also trying to make things look as much as possible like they did before, so that check can be looked at as a compatibility check. In particular, I believe that `encoding='bytes'` means 'latin1' for this PR.